src/pages/index.js

```diff
@@ -0,0 +1,303 @@
+
+const editModal = document.querySelector("#edit-modal");
+const editFormElement = editModal.querySelector("#profile__form");
+const editModalCloseBtn = editModal.querySelector(".modal__close-btn");
+const editModalNameInput = editModal.querySelector("#profile-name-input");
+const editModalDescriptionInput = editModal.querySelector(
+  "#profile-description-input"
+);
+
+const cardModal = document.querySelector("#add-card-modal");
+const cardForm = cardModal.querySelector(".modal__form");
+const cardSubmitBtn = cardModal.querySelector(".modal__submit-btn");
+const cardModalCloseBtn = cardModal.querySelector(".modal__close-btn");
+const cardNameInput = cardModal.querySelector("#add-card-name-input");
+const cardLinkInput = cardModal.querySelector("#add-card-link-input");
+
+const avatarModalButton = document.querySelector(".profile__avatar-btn");
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarForm = document.querySelector("#edit-avatar-form");
+const avatarInput = avatarForm.querySelector("#profile-avatar-input");
+const avatarModalCloseBtn = avatarModal.querySelector(".modal__close-btn");
+
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector(".modal__form");
+const deleteCancelBtn = deleteModal.querySelector(
+  ".modal__submit-btn:last-of-type"
+);
+
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+const previewModalCloseBtn = previewModal.querySelector(".modal__close-btn");
+
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+// Load user and cards
+api
+  .getAppInfo()
+  .then(([user, cards]) => {
+    currentUserId = user._id;
+    profileName.textContent = user.name;
+    profileDescription.textContent = user.about;
+    document.getElementById("profile-avatar").src = user.avatar;
+
+    cards.forEach((cardData) => {
+      cardsList.append(getCardElement(cardData));
+    });
+  })
+  .catch(console.error);
+
+// Create card DOM
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-button");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  const likes = Array.isArray(data.likes) ? data.likes : [];
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  const isLikedByUser = likes.some((user) => user._id === currentUserId);
+  if (isLikedByUser) {
+    cardLikeBtn.classList.add("card__like-button_liked");
```

Comment:
* When I reload the page, all **red** hearts become **transparent** again.


You `likes` don't exist in the `data` object. There is `isLiked`


```js

  if (data.isLiked) {
    cardLikeBtn.classList.add("card__like-button_liked");
  }
```